test/test_images/singlethreaded/main.go

```diff
@@ -18,34 +18,29 @@ limitations under the License.
 package main
 
 import (
-	"flag"
 	"fmt"
 	"net/http"
-	"sync/atomic"
 	"time"
 
+	"go.uber.org/atomic"
+
 	"knative.dev/serving/test"
 )
 
-var (
-	lockedFlag uint32
-)
+var lockedFlag = atomic.NewInt32(-1)
 
 func handler(w http.ResponseWriter, r *http.Request) {
-	// Use an atomic int as a simple boolean flag we can check safely
-	if !atomic.CompareAndSwapUint32(&lockedFlag, 0, 1) {
+	v := lockedFlag.Inc() // Returns the new value.
+	defer lockedFlag.Dec()
+	if v > 0 {
```

Comment:
nit: initialize lockedFlag with 0 and check v > 1 could be more natural.

Reply:
potato, potatoe :) I like 0 vs positive.